extra/import_data.py

```diff
@@ -1,197 +1,158 @@
+            VUS_response = None
+            while VUS_response is None:
+                try:
+                    time.sleep(5)  # 5 sec delay to request exported report.
+                    VUS_response = client.get_inheritance_analyses_variants_export(analysis['id'], export_id)
+                    #response.raise_for_status()
+                except HTTPError as exception:
+                    print(exception)
+                    pass
+            logger.info('Alissa retrieval completed of: %s' % patient_dn_no)
+            # sometimes the there are no VUS marked or found within an analysis, then no info needs to be uploaded
+            if VUS_response == []:
+                logger.info('Patient %s, has no VUS marked/found. Not uploaded to MongoDB' % patient_dn_no)
+                continue
+            else:
+                upload_to_mongodb(ONB01_analysis, accession_number, analyis_sources, patient_dn_no, VUS_response)
+    # if latest date does not retrieve any analyses from Alissa, quit program and try later
+    if analysis is None:
+        logger.info('no new analysis in Alissa since last upload to VUSualizer')
+        sys.exit(0)
+    # after uploading all analyses to MongoDB, replace time of last upload with starttime of script (before calling Alissa)
+    db.replace_one({"version" : 1}, {"lastUpdatedOn" : start_time, "version" : 1}, True)
+
+
+def upload_to_mongodb(ONB01_analysis, accession_number, analyis_sources, patient_dn_no, VUS_response):
+    '''Function, for parsing data to the MongoDB'''
+    # connection with MongoDB and the correct database "vus" and collection "variant"
+    mongoclient = pymongo.MongoClient("mongodb://localhost:27017/")
+    db = mongoclient.vus.variant
+    
+    # if DNnr already in mongoDB, check if updated an replace if neccesary
+    if db.count_documents({"dn_no": patient_dn_no}) > 0:
+        logger.info('dn_no: %s already present' % patient_dn_no)
+        # get and parse "lastUpdatedOn" from MongoDB/Alissa for specific patient_dn_no
+        lastUpdatedOn_mongoDB = db.find_one({"dn_no" : patient_dn_no}, {"lastUpdatedOn":1, "_id":0})
+        for key, value in lastUpdatedOn_mongoDB.items():
+            lastUpdatedOn_mongoDB = value
+        lastUpdatedOn_Alissa = ONB01_analysis['lastUpdatedOn']
+        if lastUpdatedOn_Alissa == lastUpdatedOn_mongoDB:
+            logger.info('%s already in database, lastUpdatedOn Alissa is the same' % patient_dn_no) 
+            return
+        elif lastUpdatedOn_Alissa > lastUpdatedOn_mongoDB:
+            db.delete_many({"dn_no": patient_dn_no})
+            logger.info('removed: %s from database, start replacing with newer version' % patient_dn_no)
+        elif lastUpdatedOn_Alissa < lastUpdatedOn_mongoDB:
+            logger.info('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
+            sys.exit('lastUpdatedOn older than within the database for %s, this should not happen' % patient_dn_no)
```

Comment:
The `sys.exit()` will exit the script, which means that 1 wrong analysis can block the update / insert of all other analyses. Is this what you want?

Reply:
I think I would make sure that the error message is picked up (monitoring log files / send error email).

Reply:
made an issue for it, to work on this problem later. I do agree with your statement, otherwise it will just be visible in the logging.
However, it is now still in the test phase, So I will make a new feature for this.